interface/main/tabs/main.php

```diff
@@ -222,13 +213,37 @@ function jsFetchGlobals(scope) {
         })
     })
 }
-
 jsFetchGlobals('top').then(globalJson => {
     jsGlobals = globalJson;
 }).catch(error => {
     console.log(error.message);
 });
 
+// set up global translations for js
+function setupI18n(lang_id) {
+    top.restoreSession();
+    return fetch(<?php echo js_escape($GLOBALS['webroot'])?> + "/library/ajax/i18n_generator.php?lang_id=" + encodeURIComponent(lang_id) + "&csrf_token_form=" + encodeURIComponent(csrf_token_js), {
+        credentials: 'same-origin',
+        method: 'GET'
+    })
+    .then(response => response.json())
+}
+setupI18n(<?php echo js_escape($_SESSION['language_choice']); ?>).then(translationsJson => {
+    i18next.init({
+        lng: 'selected',
+        debug: false,
+        nsSeparator: false,
+        keySeparator: false,
+        resources: {
+            selected: {
+                translation: translationsJson
+            }
+        }
+    });
+}).catch(error => {
+    console.log(error.message);
+});
```

Comment:
hi @sjpadgett ,
Made the changes. Testing well. Still learning js stuff, so let me know if need more work.

Reply:
btw, now gonna work on the frames removal PR which will simplify the js translation work in the future